frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/obcPage.ts

```diff
@@ -0,0 +1,89 @@
+import { MINUTE } from '../utils/consts';
+import { projectNameSpace } from '../../../dev-console/integration-tests/support/pages/app';
+import { modal } from '../../../integration-tests-cypress/views/modal';
+
+export class CreateOBCHandler {
+  name: string;
+
+  namespace: string;
+
+  storageclass: string;
+
+  bucketclass: string;
+
+  constructor(name: string, namespace: string, storageclass: string, bucketclass: string) {
+    this.name = name;
+    this.namespace = namespace;
+    this.storageclass = storageclass;
+    this.bucketclass = bucketclass;
+  }
+
+  createBucketClaim() {
+    cy.clickNavLink(['Storage', 'Object Bucket Claims']);
+    projectNameSpace.selectOrCreateProject(this.namespace);
+    cy.clickNavLink(['Storage', 'Object Bucket Claims']);
+    cy.byLegacyTestID('namespace-bar-dropdown')
+      .contains('Project')
+      .click();
+    cy.contains(this.namespace);
+    cy.byTestID('item-create').click();
+    cy.byTestID('obc-name').type(this.name);
+    cy.byTestID('sc-dropdown')
+      .should('be.visible')
+      .click();
+    cy.contains('openshift-storage.noobaa.io').click();
+    if (this.bucketclass) {
+      cy.byTestID('bc-dropdown')
+        .should('be.visible')
+        .click()
+        .contains(this.bucketclass)
+        .click();
+    }
+    modal.submit();
+  }
+
+  revealHiddenValues() {
+    cy.contains('Reveal Values').click();
+  }
+
+  hideValues() {
+    cy.contains('Hide Values').click();
+  }
+
+  assertNamespaceExists() {
+    cy.byTestSelector('details-item-value__Namespace').contains(this.namespace);
+  }
+
+  deploymentReady(deploymentName: string) {
+    cy.exec(`kubectl get deployment ${deploymentName} -o json -n ${this.namespace}`).then(
+      ($info) => {
+        const info = JSON.parse($info.stdout.toString());
+        const { replicas, readyReplicas } = info.status;
+        if (replicas < readyReplicas) {
+          cy.wait(2 * MINUTE);
+        }
+      },
+    );
```

Comment:
```suggestion
```
This is not asserting that the deployment is indeed in a ready state. Instead let's wait for it to get to ready state via UI.